woocommerce/woocommerce
# Add unit tests for Hand-Picked collection

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR adds three new test cases for the hand-picked products collection functionality:

1. `test_collection_hand_picked` - Tests that the hand-picked collection handler correctly processes selected product IDs.
2. `test_collection_hand_picked_empty` - Verifies proper handling when no products are selected.
3. `test_collection_hand_picked_order` - Ensures the original order of hand-picked products is preserved.

These tests improve coverage for the `Product Collection` block's hand-picked products feature.

Closes #51586.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Run [unit tests](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/README.md).
```
pnpm --filter=@woocommerce/plugin-woocommerce test:unit:env -- --filter=ProductCollection
```
3. All the tests should pass.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
